### ISSUE TITLE
BC NextMajor (20.0) error when Importing EVALUATION ConfigPackage "A non-empty request body is required."

### ISSUE THREAD

**Issue by Mattszczp** [microsoft/navcontainerhelper#2110]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
Hi, when trying to import a EVALUATION ConfigurationPackage to a 20.0 BC Container i get an error

Invoke-RestMethod : {"error":{"code":"BadRequest","message":"A non-empty request body is required.  CorrelationId:
9ec88820-30a0-4909-a338-7bfcf928ffa2."}}

**Scripts used to create container and cause the issue**
```
New-BCContainer -accept_eula -containerName test -multitenant:$false -artifactUrl (Get-BCArtifactUrl -select NextMajor -sasToken <sas token>) -updateHosts -auth UserPassword
UploadImportAndApply-ConfigPackageInBcContainer -containerName test
```

**Full output of scripts**
```
❯ New-BCContainer -accept_eula -containerName test -multitenant:$false -artifactUrl (Get-BCArtifactUrl -select NextMajor -sasToken <sas token>) -updateHosts -auth UserPassword
BcContainerHelper is version 2.0.15
BcContainerHelper is not running as administrator
Host is Microsoft Windows 10 Pro - 20H2
Docker Client Version is 20.10.8
Docker Server Version is 20.10.8
Downloading application artifact /sandbox/20.0.30898.0/w1
Downloading C:\Users\mateusz.szczepanski\AppData\Local\Temp\b4a7551a-ba87-4c12-be30-a7e6a5f62c04.zip
Unpacking application artifact to tmp folder using 7zip
Downloading platform artifact /sandbox/20.0.30898.0/platform
Downloading C:\Users\mateusz.szczepanski\AppData\Local\Temp\4f2bddc3-56dc-4931-86c3-ab6b0d3d78eb.zip
Unpacking platform artifact to tmp folder using 7zip
Downloading Prerequisite Components
Downloading c:\bcartifacts.cache\sandbox\20.0.30898.0\platform\Prerequisite Components\Open XML SDK 2.5 for Microsoft Office\OpenXMLSDKv25.msi
Downloading c:\bcartifacts.cache\sandbox\20.0.30898.0\platform\Prerequisite Components\IIS URL Rewrite Module\rewrite_2.0_rtw_x64.msi
Downloading c:\bcartifacts.cache\sandbox\20.0.30898.0\platform\Prerequisite Components\DotNetCore\DotNetCore.1.0.4_1.1.1-WindowsHosting.exe
Fetching all docker images
Fetching all docker volumes
Using image mcr.microsoft.com/businesscentral:10.0.19042.1237
Creating Container test
Version: 20.0.30898.0-W1
Style: sandbox
Multitenant: No
Platform: 20.0.30192.0
Generic Tag: 1.0.1.7
Container OS Version: 10.0.19042.1237 (20H2)
Host OS Version: 10.0.19042.1237 (20H2)
Using process isolation
Using locale en-US
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Files in C:\ProgramData\BcContainerHelper\Extensions\test\my:
- AdditionalOutput.ps1
- MainLoop.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container test from image mcr.microsoft.com/businesscentral:10.0.19042.1237
3700ddd18424ea73bc4e6b2531deb0cc0b9d94622c71e86f07e802374640c422
Waiting for container test to be ready
Using artifactUrl https://bcinsider.azureedge.net/sandbox/20.0.30898.0/w1
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\sandbox\20.0.30898.0\platform\ServiceTier\Program Files
c:\dl\sandbox\20.0.30898.0\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\sandbox\20.0.30898.0\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\sandbox\20.0.30898.0\platform\WindowsPowerShellScripts\WebSearch
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Determining Database Collation from c:\dl\sandbox\20.0.30898.0\w1\BusinessCentral-W1.bak
Restoring CRONUS Demo Database
Setting CompatibilityLevel for CRONUS on localhost\SQLEXPRESS
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Copying Web Client Files
c:\dl\sandbox\20.0.30898.0\platform\WebClient\Microsoft Dynamics NAV
Copying Client Files
c:\dl\sandbox\20.0.30898.0\platform\LegacyDlls\program files\Microsoft Dynamics NAV
c:\dl\sandbox\20.0.30898.0\platform\LegacyDlls\program files\Microsoft Dynamics NAV
c:\dl\sandbox\20.0.30898.0\platform\LegacyDlls\systemFolder
Copying ModernDev Files
c:\dl\sandbox\20.0.30898.0\platform
c:\dl\sandbox\20.0.30898.0\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\sandbox\20.0.30898.0\platform\ConfigurationPackages
Copying Test Assemblies
C:\dl\sandbox\20.0.30898.0\platform\Test Assemblies
Copying Extensions
C:\dl\sandbox\20.0.30898.0\w1\Extensions
Copying Applications
C:\dl\sandbox\20.0.30898.0\platform\Applications
Starting Business Central Service Tier
Importing license file
Stopping Business Central Service Tier
Installation took 84 seconds
Installation complete
Initializing...
Setting host.docker.internal to 172.18.203.173 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 172.18.203.173 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.21.240.1 in container hosts file
Starting Container
Hostname is test
PublicDnsName is test
WARNING: DNS resolution not working from within the container.
Using NavUserPassword Authentication
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 0FCA89400D978F842B2939C5AF011D2F2E4A608C
Modifying Service Tier Config File with Instance Specific Settings
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Enabling Financials User Experience
Creating http download site
Setting SA Password and enabling SA
Creating admin as SQL User and add to sysadmin
WARNING: This license is not compatible with this version of Business Central.
Creating SUPER user
WARNING: This license is not compatible with this version of Business Central.
Container IP Address: 172.21.253.0
Container Dns Name  : test
Web Client          : http://test/BC/
Dev. Server         : http://test
Dev. ServerInstance : BC
Setting test to 172.21.253.0 in host hosts file

Files:
http://test:8080/ALLanguage.vsix

Container Total Physical Memory is 15.9Gb
Container Free Physical Memory is 3.0Gb

Initialization took 107 seconds
Ready for connections!
Reading CustomSettings.config from test
Container test successfully created

Use:
Get-BcContainerEventLog -containerName test to retrieve a snapshot of the event log from the container
Get-BcContainerDebugInfo -containerName test to get debug information about the container
Enter-BcContainer -containerName test to open a PowerShell prompt inside the container
Remove-BcContainer -containerName test to remove the container again
docker logs test to retrieve information about URL's again
❯ UploadImportAndApply-ConfigPackageInBcContainer -containerName test -Credential (Get-Credential)

cmdlet Get-Credential at command pipeline position 1
Supply values for the following parameters:
Credential

cmdlet UploadImportAndApply-ConfigPackageInBcContainer at command pipeline position 1
Supply values for the following parameters:
configPackage: W1.ENU.EVALUATION
Copy from container test (C:\ConfigurationPackages\NAV20.0.W1.ENU.EVALUATION.rapidstart) to C:\ProgramData\BcContainerHelper\Extensions\test\NAV20.0.W1.ENU.EVALUATION.rapidstart
Invoke GET on http://172.21.253.0:7048/BC/api/v1.0/companies?$filter=name%20eq%20'CRONUS%20International%20Ltd.'&tenant=default
Deleting Configuration Package W1.ENU.EVALUATION (if exists)
Invoke DELETE on http://172.21.253.0:7048/BC/api/microsoft/automation/v1.0/companies(07a3f5ba-f324-ec11-bb7d-000d3a39216e)/configurationPackages('W1.ENU.EVALUATION')?tenant=default
Creating Configuration Package W1.ENU.EVALUATION
Invoke POST on http://172.21.253.0:7048/BC/api/microsoft/automation/v1.0/companies(07a3f5ba-f324-ec11-bb7d-000d3a39216e)/configurationPackages?tenant=default
Uploading Configuration Package W1.ENU.EVALUATION
Invoke PATCH on http://172.21.253.0:7048/BC/api/microsoft/automation/v1.0/companies(07a3f5ba-f324-ec11-bb7d-000d3a39216e)/configurationPackages('W1.ENU.EVALUATION')/file('W1.ENU.EVALUATION')/content?tenant=default
Importing Configuration Package W1.ENU.EVALUATION
Invoke POST on http://172.21.253.0:7048/BC/api/microsoft/automation/v1.0/companies(07a3f5ba-f324-ec11-bb7d-000d3a39216e)/configurationPackages('W1.ENU.EVALUATION')/Microsoft.NAV.import?tenant=default
Invoke-RestMethod : {"error":{"code":"BadRequest","message":"A non-empty request body is required.  CorrelationId:
9ec88820-30a0-4909-a338-7bfcf928ffa2."}}
At C:\Users\mateusz.szczepanski\OneDrive - IT.integro sp. z
o.o\Dokumenty\WindowsPowerShell\Modules\BcContainerHelper\2.0.15\Api\Invoke-NavContainerApi.ps1:156 char:5
+     Invoke-RestMethod -Method $Method -uri "$url" -Headers $headers @ ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebExc
   eption
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
```

**Screenshots**
![image](https://user-images.githubusercontent.com/12480668/135862300-9ea1b92a-1f46-4b14-ab74-32deaab5a05c.png)

**Additional context**
- does it happen all the time? Yes
- did it use to work? Yes, before BC 20.0 bacame NextMajor

**Comment by Mattszczp** [comment]
Hi Freddy, what's the status of this issue? Is there some way I could help you with solving it?

**Comment by freddydk** [comment]
Just heard from the team that this is a BC issue. They are looking into this. Will update this issue when the bug is resolved.

**Comment by freddydk** [comment]
Workaround shipped in 2.0.18
